agent/engine/docker_task_engine.go

```diff
@@ -474,7 +474,10 @@ func (engine *DockerTaskEngine) GetTaskByArn(arn string) (*api.Task, bool) {
 
 func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Container) DockerContainerMetadata {
 	pullStart := time.Now()
-	defer seelog.Infof("Finished pulling container %v. Lock acquisition and pull took %v.", container, time.Since(pullStart).String())
+	defer func(startTime time.Time) {
+		seelog.Infof("Finished pulling container %v. Lock acquisition and pull took %v.", container, time.Since(startTime).String())
```

Comment:
Shouldn't that be happening within the `concurrentPull` or the `serialPull` methods? From the perspective of the task engine, it's just `pull` that's happening. IMO, it shouldn't matter here if a lock was acquired or not as it's abstracting that.